lib/templateFactory.js

```diff
@@ -73,20 +136,74 @@ class TemplateFactory extends BaseFactory {
             });
     }
 
+    /**
+     * List templates with pagination and filter options
+     * @method list
+     * @param  {Object}   config                  Config object
+     * @param  {Object}   config.params           Parameters to filter on
+     * @param  {Object}   config.paginate         Pagination parameters
+     * @param  {Number}   config.paginate.count   Number of items per page
+     * @param  {Number}   config.paginate.page    Specific page of the set to return
+     * @return {Promise}
+     */
+    list(config) {
+        if (config.params && config.params.name && !config.params.namespace) {
+            // eslint-disable-next-line no-underscore-dangle
+            return this._getNameAndNamespace(config.params.name).then((parsedTemplateName) => {
```

Comment:
Kinda weird we are doing the same `list` inside this function then again do another `list` here

Reply:
Yeah I'm not sure if there's a better way around this